tests/functional/s3curl/tests.js

```diff
@@ -54,21 +54,18 @@ function diff(putFile, receivedFile, done) {
 
 function createFile(name, bytes, callback) {
     process.stdout.write(`dd if=/dev/urandom of=${name} bs=${bytes} count=1\n`);
-    proc.spawn('dd', ['if=/dev/urandom', `of=${name}`,
-        `bs=${bytes}`, 'count=1'], { stdio: 'inherit' }).on('exit', code => {
-            assert.strictEqual(code, 0);
-            process.stdout.write(`chmod ugoa+rw ${name}\n`);
-            proc.spawn('chmod', ['ugo+rw', name], { stdio: 'inherit' })
-                .on('exit', code => {
-                    assert.strictEqual(code, 0);
-                    callback();
-                });
-        });
+    let ret = proc.spawnSync('dd', ['if=/dev/urandom', `of=${name}`,
+        `bs=${bytes}`, 'count=1'], { stdio: 'inherit' });
+    assert.strictEqual(ret.status, 0);
+    process.stdout.write(`chmod ugoa+rw ${name}\n`);
+    ret = proc.spawnSync('chmod', ['ugo+rw', name], { stdio: 'inherit' });
+    assert.strictEqual(ret.status, 0);
+    callback();
 }
 
 function deleteFile(file, callback) {
     process.stdout.write(`rm ${file}\n`);
-    proc.spawn('rm', [file]).on('exit', () => {
+    proc.spawnSync('rm', [file]).on('exit', () => {
```

Comment:
It might  not like this async `.on` and likely you'll need to call the callback like you did in line 63